src/inject/dynamic-theme/variables.ts

```diff
@@ -310,7 +311,10 @@ export class VariablesStore {
         if (!this.typeChangeSubscriptions.has(varName)) {
             this.typeChangeSubscriptions.set(varName, new Set());
         }
-        this.typeChangeSubscriptions.get(varName).add(callback);
+        const rootStore = this.typeChangeSubscriptions.get(varName);
+        if (!rootStore.has(callback)) {
```

Comment:
I'm not sure if this is the best way, but this should prevent multiple "same" callback being added to the subscriptions.